assets/js/modules/adsense/index.js

```diff
@@ -208,6 +209,12 @@ export const ADSENSE_NOTIFICATIONS = {
 		areaSlug: NOTIFICATION_AREAS.BANNERS_BELOW_NAV,
 		viewContexts: [ VIEW_CONTEXT_MAIN_DASHBOARD ],
 		checkRequirements: async ( { select, resolveSelect } ) => {
+			const isConnected =
+				select( CORE_MODULES ).isModuleConnected( 'adsense' );
+			if ( ! isConnected ) {
+				return false;
+			}
```

Comment:
In order for this guard to work in all scenarios, we need to resolve-select the value.